js-packages/@prestojs/ui/src/UiProvider.tsx

```diff
@@ -19,9 +19,15 @@ type GetFormatterForFieldWithNull = <FieldValue, T extends HTMLElement>(
 ) => string | React.ComponentType<T> | [React.ComponentType<T>, object] | null;
 
 export interface FormItemProps {
+    children: React.ReactNode;
     required: boolean;
     help?: React.ReactNode;
     label?: React.ReactNode;
+    // Field name, if known. This is only applicable when a field has been passed through
+    // to a FormItem.
+    fieldName?: string;
+    // Any other props
+    [x: string]: any;
```

Comment:
Why is this necessary if a typescript interface is already only a minimum set of fields that must be present? Just for clarity?

Reply:
It's necessary - interface is not a minimum set of fields, eg: https://www.typescriptlang.org/play/?ssl=14&ssc=38&pln=9&pc=1#code/JYOwLgpgTgZghgYwgAgILIN4FgBQz-IhwC2EAXMgM5hSgDmA3LgL664ID2I1ycF6AXky4ChEuWQBySNUkAaEQQgAPGnADSEAJ4VJcSgBMY8lriA